tests/test_commands_cache.py

```diff
@@ -204,7 +203,15 @@ def test_update_cache(xession, tmp_path):
     assert file2.samefile(cached[basename][0])
 
 
-@skip_if_on_windows
+def test_find_binary_retains_case(tmp_path):
```

Comment:
> It's not clear to me how we ensure the casing is retained. Was it all upper case earlier?

Yes. The test would xfail on Windows prior to the fix (at b37148141cd5176d42164ea8e1ab11549924fadd). That was the failure mode of #5469.

> Can we make the test file name MiXeD case so it's clear what we're preserving,

Yes, perhaps. In df1defb7661a436c3ae47c3b061fcfe7f1010ed3, I extracted a fixture to be used across a couple of tests. Making that fixture use a name with mixed case would more explicitly show that case is being retained, but it would also cause that concern to leak across tests, which is a little awkward. e.g.

```diff
diff --git a/tests/test_commands_cache.py b/tests/test_commands_cache.py
index cc72b33f..edbce701 100644
--- a/tests/test_commands_cache.py
+++ b/tests/test_commands_cache.py
@@ -205,7 +205,7 @@ def test_update_cache(xession, tmp_path):
 
 @pytest.fixture
 def faux_binary(tmp_path):
-    binary = tmp_path / "runme.exe"
+    binary = tmp_path / "RunMe.exe"
     binary.touch()
     binary.chmod(0o755)
     return binary
@@ -214,13 +214,13 @@ def faux_binary(tmp_path):
 def test_find_binary_retains_case(faux_binary):
     cache = CommandsCache({"PATH": []})
     loc = cache.locate_binary(str(faux_binary))
-    assert "runme.exe" in loc
+    assert "RunMe.exe" in loc
 
 
 def test_exes_in_cwd_are_not_matched(faux_binary, monkeypatch):
     monkeypatch.chdir(faux_binary.parent)
     cache = CommandsCache({"PATH": []})
-    assert cache.locate_binary("runme.exe") is None
+    assert cache.locate_binary("RunMe.exe") is None
 
 
 def test_nixos_coreutils(tmp_path):
```

Note that we're not testing anything about alternate casing being passed (e.g. `locate_binary('rUNmE.exe')`) - just that if a matching case of the name is passed that it's retained. We could strengthen this guarantee, but I'm not sure it's necessary, especially while Windows and Unixes have divergent behavior.

I think you bring up a good point, though, that if there were a regression to convert everything to lowercase but the filename had uppercase characters, the test as written wouldn't catch that. So having mixed case in the filename makes sense.

Done in 602d00eb.